### PR TITLE
Add drf-yasg/swagger support

### LINKED ISSUE #307: Integrate Django REST Debugging Interface
## Expected Behavior:
- Have an option to easily switch between using templates and using the provided DRF testing interface.
## Current Behavior:
- Not implemented.

## Screenshots/ Code Snippets:
1. Something like this:
image

## Related URLs:
e.g. `https://localhost:8000`

### PR DESCRIPTION
**Pull request by mina-andajos**
Fixes #307 
----------
## Type of change:
- [x] New feature (change which adds functionality).
----------
## What's Changed:
- Added swagger-ui support, exposed a url & added [drf-yasg](https://github.com/axnsan12/drf-yasg) dependencies to `setup.py` 

--------
## Screenshots:
1. 
image

--------
## Affected URLs:
e.g. `127.0.0.1:8000/swagger/`

-------
## Checklist:
- [x] My code follows the [style guidelines](https://github.com/LesFruitsDefendus/saskatoon-ng/blob/develop/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by mina-andajos** [comment]
@2ynn hi, I tried implementing the DRF browsable API but couldn't find a helpful documentation or guide. So I thought of implementing drf-yasg instead, swagger has a very similar functionality I believe.

**Comment by 2ynn** [comment]
@mina-andajos you have to add 
` 'rest_framework.renderers.BrowsableAPIRenderer', ` to  `'DEFAULT_RENDERER_CLASSES'` in settings.py

Then if you don't provide any template_name to your view, it should fall back to the Browsable API:
```
    def retrieve(self, request, *args, **kwargs):
        ### self.template_name = 'app/detail_views/harvest/view.html'
        return super(HarvestViewset, self).retrieve(request, *args, **kwargs)
```

**Comment by 2ynn** [comment]
@mina-andajos swagger is nice for documentation and testing but I find using filters and links in the default Rest Browser API more convenient for developing.

File: saskatoon/saskatoon/urls.py [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
+    openapi.Info(
+        title="Snippets API",
+        default_version='v1',
+        description="Test description",
+        terms_of_service="https://www.google.com/policies/terms/",
+        contact=openapi.Contact(email="contact@snippets.local"),
+        license=openapi.License(name="BSD License"),
+    ),
+    public=True,
+    permission_classes=[permissions.AllowAny],
```
Comment (2ynn) [inline saskatoon/saskatoon/urls.py:19]: @mina-andajos let's at least enforce `IsAuthenticated` permission.
Alternatively we could also write a custom `IsAdmin` permission, similar to `IsCoreOrAdmin` in `harvest.permissons`

**Review by 2ynn — CHANGES_REQUESTED** [review]

**Comment by mina-andajos** [comment]
I'm closing this, changing branch and implementing the drf browsable api instead of swagger

**Pull request closed without merging**